NewEmuChecklist.txt

```diff
@@ -0,0 +1,75 @@
+  an additional core is much easier than overhauling an entire emulator.
+
+[ ] Game Identification
+  [ ] Hash method documented and approved
+  [ ] If emulator supports dynamic patching [i.e. ips], hash is calculated after applying patch
+  [ ] Hash ignores any persistant save data [SRAM]
+  [ ] Game can be properly identified (RA_OnLoadNewRom)
+  [ ] Unidentified game provides some descriptor [typically filename] to unknown game dialog (RA_InstallSharedFunctions>RA_GetEstimatedGameTitle)
+  [ ] Unidentified game provides list of known games for console (RA_SetConsoleID)
+[ ] Emulator
+  [ ] DLL is downloaded if not available (RA_Init - new enum may be needed)
+  [ ] RetroAchievements menu is visible (RA_CreatePopupMenu) and functional (RA_InvokeDialog)
+  [ ] RetroAchievements menu updates when logging in/out (RA_InstallSharedFunctions>RA_RebuildMenu)
+  [ ] Login dialog displayed on first run or auto-login on future runs (RA_AttemptLogin)
+  [ ] Title includes logged in user (RA_UpdateAppTitle)
+  [ ] RALog.txt ends with "Shutdown complete" after closing emulator (RA_Shutdown)
+  [ ] Full-window overlay can be displayed (RA_SetPaused)
+  [ ] Full-window overlay can be navigated (RA_UpdateRenderOverlay>pInput)
+  [ ] Full-window overlay navigation uses player's custom controller bindings, not hard-coded inputs.
+  [ ] Closing the full-window overlay unpauses the emulator (RA_InstallSharedFunctions>RA_CauseUnpause)
+  [ ] Resetting the emulator disables any active popups/leaderboards (RA_OnReset)
+  [ ] Performance of emulator (without any debugging windows open) should not be noticably different than the source emulator
+  [ ] Emulator must perform at intended framerate on an "average" machine.
+[ ] Achievements/Leaderboards/Rich Presence
+  [ ] Achievements can be created (RA_InstallSharedFunctions>RA_GameIsActive)
+  [ ] Achievements trigger (RA_InstallMemoryBank>pReader, RA_DoAchievementsFrame, RA_HandleHTTPResults)
+  [ ] Achievement trigger displays notification (RA_UpdateRenderOverlay)
+  [ ] Achievement can pause emulator [Pause on Reset, Pause on Trigger] (RA_InstallSharedFunctions>RA_CausePause)
+  [ ] Achievements are not processed when emulator is paused [HitCount for true condition does not increment]
+  [ ] When using frame advance, achievements are only processed when the frame is advanced.
+  [ ] Changing the current game prompts the user to save modifications (RA_ConfirmLoadNewRom)
+    [ ] Choosing not to save modifications aborts changing the current game.
+  [ ] Closing the emulator prompts the user to save modifications (RA_ConfirmLoadNewRom)
+    [ ] Choosing not to save modifications aborts closing the emulator.
+  [ ] Closing the current game prompts the user to save modifications (RA_ConfirmLoadNewRom)
+    [ ] Choosing not to save modifications aborts closing the current game.
+  [ ] Closing the current game clears the list of achievements for the game (RA_OnLoadNewRom(NULL,0))
+[ ] Softcore
```

Comment:
In general, functionality that is allowed in softcore should not need to be explicitly specified in my opinion, as long as the `Hardcore` section is clear.

Reply:
You'd think so, but if the hardcore check is not implemented correctly, it causes issues in softcore. Having the list just makes sure the functionality is tested.